### PR TITLE
Standardize messages that suggest to run a command-line command

### PR DESCRIPTION
**Pull request by eksperimental**
It tries to avoid the use of different styles such as:
- inline commands sentence with user of double-quotes (ie. Invalid arguments, please run "mix something" for more information)
- inline commands in sentence with use of semi-colon (ie. Invalid arguments, please run: mix something)

Suggested commands go in a new line, indented with 4
white-spaces.

Error messages follow the following format:

```
** (Mix) Invalid arguments given to "mix local.rebar"

Check the proper call syntax with:
    mix help local.rebar
```

### CODE REVIEW

**Comment by whatyouhide** [comment]
I am not sure about this PR. I personally don't see the discrepancies as something that justifies standardization as all versions are understandable, but I understand the rationale behind standardizing. However, before standardizing a discussion on what the best standard is plus a guideline written somewhere on this are in order in my opinion

**Comment by josevalim** [comment]
I think having a standard is good but not the one that was chosen exactly because they are embedded in other messages so spawning over multiple lines is not desirable. If we are going to pick a format, we should go with `please run "mix deps.get"` or `run "mix deps.get"` instead. The exception is for long commands, which then breaking into a new line makes sense, but I am :-1: on making all messages spawn multiple lines by default.

TL;DR: we can change the `do this: ...` into `do this "..."`. The long messages can stay as is. :)

**Comment by eksperimental** [comment]
I agree with @josevalim, but I think when we raise it is good to have a line with the description of the error.
And I think it is desireable that if we want to give suggestions it goes on a new line as in my examples. I could be with a different format as the one José suggests, the problem with the double quotes is that is is confusing when we need to pass an argument wrapped in quotes (it may not be necessary for the Elixir project itself, but it could be the case for other projects, and it will be nice if one standard can be used across all apps).

**Comment by josevalim** [comment]
I don't think there is a need to generalize across all applications. Making it generic may mean we end-up with a format that is non-ideal for everyone, instead of those apps using what makes sense to them. For example, what makes sense for most of Mix is the double quotes format, so that's what we should go with.

**Comment by eksperimental** [comment]
@josevalim so do you want the suggestion in a different line when we we Mix.raise?

**Comment by josevalim** [comment]
@eksperimental unless the example may have quotes in it, let's keep everything on a single line as it is today.

File: lib/mix/lib/mix/dep.ex [hunk 1]
```diff
@@ -209,16 +209,23 @@ defmodule Mix.Dep do
     do: "the dependency does not match the requirement #{inspect req}, got #{inspect vsn}"
 
   def format_status(%Mix.Dep{status: {:lockmismatch, _}}),
-    do: "lock mismatch: the dependency is out of date (run \"mix deps.get\" to fetch locked version)"
```
Comment (eksperimental) [inline lib/mix/lib/mix/dep.ex]: should we have a common format here?, such as: 
> "lock mismatch: the dependency is out of date. To fetch locked version run: \"mix deps.get\""

File: lib/mix/lib/mix/local/installer.ex [hunk 2]
```diff
@@ -113,17 +117,11 @@ defmodule Mix.Local.Installer do
...
-          #{message}
-
-          Could not fetch #{name} at:
-
-              #{src}
-
-          Please download the #{name} above manually to your current directory and run:
-
-              mix #{name}.install ./#{src_basename}
-          """
```
Comment (eksperimental) [inline lib/mix/lib/mix/local/installer.ex]: @josevalim: do you want to do anything about this one that is broken in different lines?

File: lib/mix/lib/mix/tasks/local.rebar.ex [hunk 3]
```diff
@@ -73,17 +74,11 @@ defmodule Mix.Tasks.Local.Rebar do
...
-          #{message}
-
-          Could not fetch #{manager} at:
-
-              #{path}
-
-          Please download the file above manually to your current directory and run:
-
-              mix local.rebar #{manager} ./#{Path.basename(local)}
-          """
```
Comment (eksperimental) [inline lib/mix/lib/mix/tasks/local.rebar.ex]: @josevalim and this one?

File: lib/mix/lib/mix/local/installer.ex (hunk 2 above)
Comment (josevalim) [inline lib/mix/lib/mix/local/installer.ex]: This one is fine as is.

File: lib/mix/lib/mix/tasks/local.rebar.ex (hunk 3 above)
Comment (josevalim) [inline lib/mix/lib/mix/tasks/local.rebar.ex]: Same.

File: lib/mix/lib/mix/dep.ex (hunk 1 above)
Comment (josevalim) [inline lib/mix/lib/mix/dep.ex]: Let's go with:

> To fetch locked version run "mix deps.get"

Let's remove the `:` from all of those as it only works if the command is the last thing on the sentence.

**Comment by eksperimental** [comment]
New approach submitted, please have a look

File: lib/mix/lib/mix/dep.ex [hunk 4]
```diff
@@ -209,13 +209,13 @@ defmodule Mix.Dep do
...
+    do: "lock mismatch: the dependency is out of date. To fetch locked version run \"mix deps.get\""
 
   def format_status(%Mix.Dep{status: :lockoutdated}),
-    do: "lock outdated: the lock is outdated compared to the options in your mixfile (run \"mix deps.get\" to fetch locked version)"
+    do: "lock outdated: the lock is outdated compared to the options in your mixfile. To fetch locked version run \"mix deps.get\""
 
   def format_status(%Mix.Dep{status: :nolock}),
-    do: "the dependency is not locked (run \"mix deps.get\" to generate \"mix.lock\" file)"
+    do: "the dependency is not locked. To generate \"mix.lock\" file run \"mix deps.get\""
```
Comment (whatyouhide) [inline lib/mix/lib/mix/dep.ex]: This doesn't feel right, I think it's "to generate **the** mix.lock ...", wdyt?

Comment (eksperimental) [inline lib/mix/lib/mix/dep.ex]: Yes, but if so we shouldn't use quotes, so it should be:
> To generate the mix.lock file run \"mix deps.get\"

Comment (whatyouhide) [inline lib/mix/lib/mix/dep.ex]: It should be

> To generate the "mix.lock" file run "mix deps.get"

Comment (eksperimental) [inline lib/mix/lib/mix/dep.ex]: yes, you are right. the standard is to use double quotes for filenames and paths. fixed!

File: lib/mix/lib/mix/dep.ex [hunk 5]
```diff
@@ -209,13 +209,13 @@ defmodule Mix.Dep do
...
-    do: "lock mismatch: the dependency is out of date (run \"mix deps.get\" to fetch locked version)"
+    do: "lock mismatch: the dependency is out of date. To fetch locked version run \"mix deps.get\""
 
   def format_status(%Mix.Dep{status: :lockoutdated}),
-    do: "lock outdated: the lock is outdated compared to the options in your mixfile (run \"mix deps.get\" to fetch locked version)"
+    do: "lock outdated: the lock is outdated compared to the options in your mixfile. To fetch locked version run \"mix deps.get\""
 
   def format_status(%Mix.Dep{status: :nolock}),
-    do: "the dependency is not locked (run \"mix deps.get\" to generate \"mix.lock\" file)"
+    do: "the dependency is not locked. To generate the \"mix.lock\" file run \"mix deps.get\""
```
Comment (whatyouhide) [inline lib/mix/lib/mix/dep.ex]: I think the test for this error needs to be updated as well.

Comment (eksperimental) [inline lib/mix/lib/mix/dep.ex]: oops. fixed!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
